### PR TITLE
[firebase_storage] Support Android v2 embedding

### PR DESCRIPTION
**Pull request by kroikie**
Fixes: https://github.com/flutter/flutter/issues/41877

### CODE REVIEW

File: packages/firebase_storage/example/pubspec.yaml [hunk 1]
```diff
@@ -10,6 +10,7 @@ dependencies:
   firebase_core: ^0.4.0
   uuid: ^1.0.0
   http: ^0.12.0
+  e2e: ^0.2.1
```
Comment (xster) [inline packages/firebase_storage/example/pubspec.yaml]: should this be a dev_dependency?

Comment (kroikie) [reply packages/firebase_storage/example/pubspec.yaml]: Done!

**Comment by xster** [comment]
Reminder to sync and merge this

**Comment by kroikie** [comment]
Once the build is green, I'll rebase and merge this PR.

**Pull request merged**